src/adapters/ip_camera/api.rs

```diff
@@ -104,6 +124,75 @@ impl IpCamera {
         Ok(camera)
     }
 
+    fn config_key(&self, key: &str) -> String {
+        format!("{}.{}", self.udn, key)
+    }
+
+    fn get_config(&self, key: &str) -> Option<String> {
+        self.config.get("ip_camera", &self.config_key(key))
+    }
+
+    fn set_config(&self, key: &str, value: &str) {
+        self.config.set("ip_camera", &self.config_key(key), value);
+    }
+
+    pub fn get_name(&self) -> String {
+        if let Some(name) = self.get_config("name") {
+            info!("get_name got '{}'", name);
+            if !name.is_empty() {
+                info!("get_name return config name '{}'", name);
+                return name;
+            }
+        }
+        info!("get_name return upnp name '{}'", self.upnp_name);
+        self.upnp_name.clone()
+    }
+
+    pub fn set_name(&self, name: &str) {
+        self.set_config("name", name);
+    }
+
+    pub fn get_location(&self) -> String {
```

Comment:
As mentioned in the other PR, I would prefer if we could put locations in tags, with the convention: `"location: foobar"`. Would this be feasible?